app/code/Magento/Swatches/Model/Plugin/EavAttribute.php

```diff
@@ -174,8 +174,10 @@ protected function prepareOptionIds(array $optionsArray)
     {
         if (isset($optionsArray['value']) && is_array($optionsArray['value'])) {
             foreach (array_keys($optionsArray['value']) as $optionId) {
-                if (isset($optionsArray['delete']) && $optionsArray['delete'][$optionId] == 1) {
-                    unset($optionsArray['value'][$optionId]);
+                if (isset($optionsArray['delete']) && isset($optionsArray['delete'][$optionId])) {
+                    if ($optionsArray['delete'][$optionId]== 1) {
```

Comment:
empty will consume more cpu cycle, and is not equivalent to isset. the purpose of this check is to avoid the notice, and it will. empty will doone more thing, it will skip the "unset" if $optionId is set to ''. 
IMHO magento 2 is slow enough to not use "simpler condition" that is actually more expensive ;)